__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -151,6 +248,172 @@ describe('start', () => {
     wallet.setGapLimit(originalGapLimit);
     hWallet.stop();
   });
+
+
+  it('should start a wallet to manage a specific token', async () => {
+    const walletData = precalculationHelpers.test.getPrecalculatedWallet();
+
+    // Creating a new wallet with a known set of words just to generate the custom token
+    let hWallet = await generateWalletHelper({
+      seed: walletData.words,
+      preCalculatedAddresses: walletData.addresses,
+    });
+    await GenesisWalletHelper.injectFunds(hWallet.getAddressAtIndex(0), 2);
+    const { hash: tokenUid } = await createTokenHelper(
+      hWallet,
+      'Dedicated Wallet Token',
+      'DWT',
+      100
+    );
+
+    // Stopping this wallet and destroying its memory state
+    await hWallet.stop({ cleanStorage: true });
+    hWallet = null;
+
+    // Starting a new wallet re-using the same words, this time with a specific wallet token
+    hWallet = await generateWalletHelper({
+      seed: walletData.words,
+      preCalculatedAddresses: walletData.addresses,
+      tokenUid,
+    });
+    expect(hWallet.isReady()).toStrictEqual(true); // This operation should work
+
+    // Now testing the methods that use this set tokenUid information
+    // FIXME: No need to explicitly pass the non-boolean `false` as a tokenUid to get this result.
+    await expect(await hWallet.getBalance(false)).toStrictEqual([
+      {
+        token: {
+          id: tokenUid,
+          name: "",
+          symbol: ""
+        },
+        balance: {
+          unlocked: 100,
+          locked: 0
+        },
+        transactions: 3,
+        lockExpires: null,
+        tokenAuthorities: {
+          unlocked: {
+            mint: true,
+            melt: true
+          },
+          locked: {
+            mint: false,
+            melt: false
+          }
+        }
+      }
+    ]);
+
+    // FIXME: We should not have to explicitly pass an empty token uid to get this result
+    const txHistory1 = await hWallet.getTxHistory({ token_id: undefined });
+    expect(txHistory1).toStrictEqual([
+      expect.objectContaining({
+        txId: tokenUid,
+        tokenUid: tokenUid,
+        balance: 100
+      }),
+    ]);
+
+    /*
+     * These tests could be created inside the `getBalance` and `getTxHistory` sections but for
+     * simplicity sake, since they are so small, were added here just as a complement to
+     * this `start` test.
+     */
+  });
+
+  it('should start a wallet via xpub', async () => {
+    const walletData = precalculationHelpers.test.getPrecalculatedWallet();
+    const xpriv = walletUtils.getXPrivKeyFromSeed(
+      walletData.words,
+      { networkName: NETWORK_NAME })
+    const privkey = walletUtils.deriveXpriv(xpriv, "0'/0");
+    const xpub = privkey.xpubkey;
+
+    // Creating a new wallet with a known set of words just to generate the custom token
+    let hWallet = await generateWalletHelper({ xpub });
```

Comment:
Seeing the code below

```
// Start the wallet
  const walletConfig = {
    seed: walletData.words,
    connection: generateConnection(),
    password: DEFAULT_PASSWORD,
    pinCode: DEFAULT_PIN_CODE,
    preCalculatedAddresses: walletData.addresses,
  };
  if (param) {
    Object.assign(walletConfig, param);
  }
```

It looks like that this wallet will be started with xpub but also with pin/password, what is not correct in the case of an xpub wallet because we use the pin only for actions to access the private key. We must change this to start the wallet only with the xpub and everything should work fine.

Reply:
You're right, this was being instantiated with pin and password.
Removed both these parameters in [b1b919e](https://github.com/HathorNetwork/hathor-wallet-lib/pull/409/commits/b1b919ea17d12af980db07820fb083bcf31118c7).